src/aktualizr_secondary/aktualizr_secondary.h

```diff
@@ -1,50 +1,62 @@
 #ifndef AKTUALIZR_SECONDARY_H
 #define AKTUALIZR_SECONDARY_H
 
-#include <memory>
-
-#include "aktualizr_secondary_common.h"
 #include "aktualizr_secondary_config.h"
-#include "aktualizr_secondary_interface.h"
-#include "crypto/keymanager.h"
-#include "socket_server.h"
-#include "storage/invstorage.h"
-#include "utilities/types.h"
-#include "utilities/utils.h"
+#include "aktualizr_secondary_metadata.h"
+#include "uptane/secondaryinterface.h"
 
 #include "uptane/directorrepository.h"
 #include "uptane/imagesrepository.h"
-#include "uptane/tuf.h"
+#include "uptane/manifest.h"
 
-#include "aktualizr_secondary_metadata.h"
+class UpdateAgent;
+class INvStorage;
+class KeyManager;
 
-class AktualizrSecondary : public AktualizrSecondaryInterface, private AktualizrSecondaryCommon {
+class AktualizrSecondary : public Uptane::SecondaryInterface {
  public:
-  AktualizrSecondary(const AktualizrSecondaryConfig& config, const std::shared_ptr<INvStorage>& storage);
-  void run() override;
-  void stop() override;
-
-  // implementation of primary's SecondaryInterface
-  Uptane::EcuSerial getSerialResp() const;
-  Uptane::HardwareIdentifier getHwIdResp() const;
-  PublicKey getPublicKeyResp() const;
-  Json::Value getManifestResp() const;
-  bool putMetadataResp(const Metadata& metadata);
-  int32_t getRootVersionResp(bool director) const;
-  bool putRootResp(const std::string& root, bool director);
-  bool sendFirmwareResp(const std::shared_ptr<std::string>& firmware);
-
-  static void extractCredentialsArchive(const std::string& archive, std::string* ca, std::string* cert,
-                                        std::string* pkey, std::string* treehub_server);
+  using Ptr = std::shared_ptr<AktualizrSecondary>;
+
+ public:
+  // TODO: free AktualizrSecondary from dependencies as much as possible, e.g. bootloader
```

Comment:
It has now been freed of the bootloader, at least from the perspective of the constructor here. :)

Reply:
yes